engines/router/missionctl/log/resultlog/proto/turing/TuringResultLog.proto

```diff
@@ -5,46 +5,42 @@ package turing;
 option java_outer_classname = "TuringResultLogProto";
 option go_package = "github.com/gojek/turing";
 
-import "google/protobuf/struct.proto";
 import "google/protobuf/timestamp.proto";
 
 message Request {
-    // The header from the incoming request to the Turing router
-    map<string, google.protobuf.ListValue> header = 1;
+    // The header from the incoming request to the Turing router. The map value is a comma-delimited string.
+    map<string,string> header = 1;
 
-    // The JSON body of the request to the Turing router
-    google.protobuf.Struct body = 2;
+    // The JSON body of the request to the Turing router, UTF-8-encoded
+    string body = 2;
 }
 
 message Response {
-    // The response from a Turing component (Enricher / Experiment Engine / Router / Ensembler).
-    google.protobuf.Struct response = 1;
-    
+    // The JSON response from a Turing component (Enricher / Experiment Engine / Router / Ensembler), UTF-8-encoded.
+    string response = 1;
+
     // The error from a Turing component, when a successful response is not received.
     string error = 2;
```

Comment:
Is this error always non-empty when the respose is not ok? i.e if both response and error is empty string, means it is a successful response with no error rather than erronous response with no error detail.

Reply:
Yes, this error will always be non-empty when the response is empty or not 2XX ([router](https://github.com/gojek/turing/blob/main/engines/router/missionctl/mission_control.go#L229), [enrich/ensemble](https://github.com/gojek/turing/blob/main/engines/router/missionctl/mission_control.go#L154)) since Turing wraps the error received with more info / creates a new error object.